sdcm/mgmt.py

```diff
@@ -4,11 +4,234 @@
+        if result.stderr:
+            raise ScyllaManagerError("Failed getting Manager version: {}".format(result.stderr))
+        logger.info("Manager version is: {}".format(result.stdout))
+        return result.stdout
+
+    def wait_for_task_status(self, task_id, list_status, cluster_id, check_task_progress=False):
+        text = "Waiting until task: {} reaches status of: {}".format(task_id, list_status)
+        is_status_reached = False
+        is_status_reached = wait.wait_for(func=self.is_task_status_in_list, step=60,
+                                          text=text, task_id=task_id, list_status=list_status, cluster_id=cluster_id, check_task_progress=check_task_progress, timeout=3600)
+        return is_status_reached
+
+    def is_task_status_in_list(self, task_id, cluster_id, list_status, check_task_progress=False):
+        """
+        Check if the status of a given task is in list
+        :param task_id:
+        :param cluster_id:
+        :param list_status:
+        :return:
+        """
+        if check_task_progress:
+            self.get_task_progress(task_id=task_id, cluster_id=cluster_id)
+        return self.get_task_status(task_id=task_id, cluster_id=cluster_id) in list_status
+
+    def wait_for_task_status_done(self, task_id, cluster_id):
+        """
+        Wait and report that task is eventually reaching a 'final' status. meaning one of: done/error/stopped
+        :param task_id:
+        :param cluster_id:
+        :return:
+        """
+        cur_status = self.get_task_status(task_id=task_id, cluster_id=cluster_id)
+        if cur_status == TaskStatus.DONE:
+            return True
+        if cur_status in [TaskStatus.ERROR, TaskStatus.STOPPED]:
+            return False
+
+        if cur_status == TaskStatus.NEW:
+            list_status = [TaskStatus.RUNNING, TaskStatus.ERROR, TaskStatus.DONE]
+            res = self.wait_for_task_status(task_id=task_id, cluster_id=cluster_id, list_status=list_status)
+            if not res:
+                raise ScyllaManagerError("Unexpected result on waiting for task {} status".format(task_id))
+            cur_status = self.get_task_status(task_id=task_id, cluster_id=cluster_id)
+            if cur_status == TaskStatus.ERROR:
+                return False
+
+        if cur_status == TaskStatus.RUNNING:
+            list_status = [TaskStatus.DONE, TaskStatus.ERROR]
+            res = self.wait_for_task_status(task_id=task_id, cluster_id=cluster_id, list_status=list_status, check_task_progress=True)
+            if not res:
+                raise ScyllaManagerError("Unexpected result on waiting for task {} status {}".format(task_id, list_status))
+            cur_status = self.get_task_status(task_id=task_id, cluster_id=cluster_id)
+            if cur_status == TaskStatus.ERROR:
+                return False
+
+        return cur_status == TaskStatus.DONE
+
+    def _mgr_remoter_run(self, cmd):
+        result = self.manager_node.remoter.run(cmd)
```

Comment:
you don't need this method, you can just pass `verbose=True` to `remoter.run`

Reply:
done